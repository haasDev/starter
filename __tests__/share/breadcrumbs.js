import React from 'react';
import Breadcrumbs from '../../src/share/breadcrumbs';
import {mount} from 'enzyme';
import snapshot from '../snapshot';

describe('Breadcrumbs', () => {
  const items = [
    {id: 1, label: 'foo'},
    {id: 2, label: 'bar'},
    {id: 3, label: 'baz'}
  ];
  const onNavigate = jest.fn();
  const jsx =
    <Breadcrumbs
      activeCrumb={2}
      items={items}
      onNavigate={onNavigate}
    />;

  it('should render', () => {
    snapshot(jsx);
  });

  it('should handle clicks', () => {
    const wrapper = mount(jsx);

    const links = wrapper.find('.breadcrumb a');
    // links is an object, not an array!
    // The last crumb is not a link, so we expect 2, not 3.
    expect(links.length).toBe(2);

    const barLink = links.at(1);
    barLink.simulate('click');
    expect(onNavigate).toBeCalled();
  });
});
