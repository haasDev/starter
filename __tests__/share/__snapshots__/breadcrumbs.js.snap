exports[`Breadcrumbs should render 1`] = `
<ol
  aria-label="breadcrumbs"
  className="breadcrumb"
  role="navigation">
  <li
    className="">
    <a
      href="#"
      onClick={[Function]}
      role="button"
      target={undefined}
      title={undefined}>
      foo
    </a>
  </li>
  <li
    className="active">
    <span>
      bar
    </span>
  </li>
  <li
    className="">
    <a
      href="#"
      onClick={[Function]}
      role="button"
      target={undefined}
      title={undefined}>
      baz
    </a>
  </li>
</ol>
`;
